conjure-runtime/src/service/http_error.rs

```diff
@@ -74,150 +70,85 @@ pub struct HttpErrorService<S> {
+                };
+
+                Err(e)
+            }
+            _ => {
+                let (parts, body) = response.into_parts();
+
+                let body = match body::to_bytes(Limited::new(body, 10 * 1024)).await {
+                    Ok(body) => body,
+                    Err(e) => {
+                        info!("error reading response body", error: Error::internal(e));
+                        Bytes::new()
                     }
-                }
-                Projection::ReadingBody {
-                    status,
-                    mut body,
-                    buf,
-                    service_error,
-                } => {
-                    loop {
-                        let data = match ready!(body.as_mut().poll_data(cx)) {
-                            Some(Ok(data)) => data,
-                            Some(Err(e)) => {
-                                info!("error reading response body", error: Error::internal(e));
-                                break;
-                            }
-                            None => break,
-                        };
-
-                        buf.put(data);
-                        // limit how much we read in case something weird is going on
-                        if buf.len() > 10 * 1024 {
-                            break;
-                        }
+                };
+
+                let error = RemoteError {
+                    status: parts.status,
+                    error: json::client_from_slice(&body).ok(),
+                };
+                let log_body = error.error.is_none();
+
+                let mut error = match (&error.error, self.service_error) {
+                    (Some(e), ServiceError::PropagateToCaller) => {
+                        let e = e.clone();
+                        Error::propagated_service_safe(error, e)
                     }
-
-                    let error = RemoteError {
-                        status: *status,
-                        error: json::client_from_slice(buf).ok(),
-                    };
-                    let log_body = error.error.is_none();
-                    let mut error = match (&error.error, service_error) {
-                        (Some(e), ServiceError::PropagateToCaller) => {
-                            let e = e.clone();
-                            Error::propagated_service_safe(error, e)
-                        }
-                        (Some(_), ServiceError::WrapInNewError) | (None, _) => {
-                            Error::internal_safe(error)
-                        }
-                    };
-                    if log_body {
-                        error = error.with_unsafe_param("body", String::from_utf8_lossy(buf));
+                    (Some(_), ServiceError::WrapInNewError) | (None, _) => {
+                        Error::internal_safe(error)
                     }
+                };
 
-                    return Poll::Ready(Err(error));
+                if log_body {
```

Comment:
nit: I think we can avoid the `mut` on `error`
```rust
let error = if log_body { error.with_unsafe_param(....) } else { error }
```

Reply:
That's true, but IMO it's a bit more confusing.